src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.cs

```diff
@@ -191,18 +191,6 @@ public ImmutableArray<StateSet> CreateBuildOnlyProjectStateSet(Project project)
                 return stateSets.ToImmutable();
             }
 
-            public bool OnProjectRemoved(IEnumerable<StateSet> stateSets, ProjectId projectId)
-            {
-                var removed = false;
-                foreach (var stateSet in stateSets)
-                {
-                    removed |= stateSet.OnProjectRemoved(projectId);
-                }
-
-                _projectAnalyzerStateMap.TryRemove(projectId, out _);
-                return removed;
-            }
```

Comment:
note: we likely need to have the incremental analyzer take on this responsibility in some fashion.  In a follow up i think we can do what we do elsewhere and do a simple listening to workspace events with an AsyncBatchingWorkQueue, and when we hear about project/doc deletinos, we can clear our state.